### DISCUSSION TITLE
Example Request: Unwrapped opus packets to webserver

### DISCUSSION THREAD

**Discussion by elshnkhll** [pschatzmann/arduino-audio-tools#1013] [Q&A] [answered]
### Problem Description

Hi Phil,

I am receiving plain (not wrapped in any container like ogg or mp3) Opus (48000 Hz, 2 channel) UDP packets every 20ms on ESP32 port. Would it be possible to wrap them using ContainerOgg and play them in a browser using this amazing library? Example would be greatly appreciated.

Thanks and Best Wishes.

### Device Description

ESP32

### Sketch

```cpp
no code
```

### I have checked existing issues, discussions and online documentation

- [X] I confirm I have checked existing issues, discussions and online documentation

**Comment by pschatzmann** [comment, marked as answer]
I don't quite understand why you report this as a bug, so I convert it into a discussion and I don't see any reason why this should not be possible. 

Just read the Codec documentation provided in the Wiki and have a look at the corresponding examples and test cases. Usually you would encode PCM data, but if you already have Opus packets, I guess you could use the OggContainerEncoder or OggContainerOutput class to wrap them into Ogg and write the data into a QueuStream. This would then serve as data source for the AudioServer.

**Comment by elshnkhll** [comment]
Thanks for quick response. Those packets are available inside of UDP event handler function. How do I convert them into audio source?

**Reply by pschatzmann** [reply]
As mentioned above you would write them into a QueuStream
